wandb/sdk/lib/ipython.py

```diff
@@ -13,13 +14,29 @@
 """
 
 
+def _is_jupyter_process():
+    """
+    Detect if a script is run as a sub process (!python train.py) in a jupyter environment.
+    The script could start its own sub procs and this method would return True for all of them.
+    """
+    p = psutil.Process()
+    while p:
+        if p.name().startswith("jupyter-note"):
```

Comment:
What makes us sure all ipython sessions will have this name?

Reply:
Looking at better solutions...

Reply:
Done.